torch/distributed/tensor/_random.py

```diff
@@ -198,7 +201,7 @@ def _distribute_region(self, spec: DTensorSpec):
         if self.distribute_region_enabled:
             old_offset = self.get_offset("parallel-rng")
             self._set_pre_op_offset(spec)
-            with torch.random.fork_rng(devices=[self._device]):
+            with torch.random.fork_rng(devices=[self._device], device_type=self._device.type):
```

Comment:
Now, in `torch.random.fork_rng()`, `device_type` defaults to `cuda` even if we pass `devices` with another type. So, maybe `device_type` should default to the type of `devices` instead of a hard-coded type?

https://github.com/pytorch/pytorch/blob/c1dd75e4dcd354a3d20e2ebb668b1ab8221a4973/torch/random.py#L125-L132